modules/features/filters/src/main/java/au/com/shiftyjelly/pocketcasts/filters/FilterEpisodeListFragment.kt

```diff
@@ -438,35 +437,34 @@ class FilterEpisodeListFragment : BaseFragment() {
 
         val multiSelectToolbar = binding.multiSelectToolbar
         multiSelectHelper.source = SourceView.FILTERS
-        multiSelectHelper.isMultiSelectingLive.observe(
-            viewLifecycleOwner,
-            Observer {
-
-                if (!multiSelectLoaded) {
-                    multiSelectLoaded = true
-                    return@Observer // Skip the initial value or else it will always hide the filter controls on load
-                }
+        multiSelectHelper.isMultiSelectingLive.observe(viewLifecycleOwner) { isMultiSelecting ->
+            if (!multiSelectLoaded) {
+                multiSelectLoaded = true
+                return@observe // Skip the initial value or else it will always hide the filter controls on load
+            }
 
+            val wasMultiSelecting = multiSelectToolbar.isVisible
+            if (wasMultiSelecting != isMultiSelecting) {
                 analyticsTracker.track(
-                    if (it) {
+                    if (isMultiSelecting) {
                         AnalyticsEvent.FILTER_MULTI_SELECT_ENTERED
                     } else {
                         AnalyticsEvent.FILTER_MULTI_SELECT_EXITED
                     }
                 )
+            }
 
-                if (!multiSelectToolbar.isVisible) {
-                    showingFilterOptionsBeforeMultiSelect = layoutFilterOptions.isVisible
-                    setShowFilterOptions(false)
-                } else {
-                    setShowFilterOptions(showingFilterOptionsBeforeMultiSelect)
-                }
-                multiSelectToolbar.isVisible = it
-                toolbar.isVisible = !it
-
-                adapter.notifyDataSetChanged()
+            if (isMultiSelecting) {
```

Comment:
It seemed strange this was using the old multi-select state instead of what it needed to be.